node/commons/products.ts

```diff
@@ -13,15 +14,35 @@ interface ConvertProductInput {
 export const productsBiggy = async ({ searchResult, ctx, simulationBehavior = 'default', tradePolicy, regionId }: ConvertProductInput) => {
   const { segment } = ctx.vtex
   const checkout = ctx.clients.checkout
-  const products: any[] = []
+  let products: any[] = []
 
-  searchResult.products.forEach((product: any) => {
-    try {
-      products.push(convertBiggyProduct(product, checkout, simulationBehavior, segment, tradePolicy ?? segment?.channel, regionId))
```

Comment:
isn't it already running in parallel?

Reply:
Per our off-line talk, yes. All premises get resolved by GraphQL, so they all happen in parallalel.